torch/_inductor/codegen/triton.py

```diff
@@ -114,14 +117,14 @@ class OpDtypeSupport:
     convert_outputs: dict[str, bool] = {}
 
     @classmethod
-    def register_upcast(cls, func: Callable[..., str], convert_output: bool):
+    def register_upcast(cls, func: Callable[..., str], convert_output: bool) -> None:
```

Comment:
Nit: 
```suggestion
    def register_upcast(cls, func: Callable[_P, str], convert_output: bool) -> None:
```
Where _P is a typing_extensions.ParamSpec, although probably doesn't matter in this case

Reply:
Doesn't this only matter if it is used in the return value (or another input)?

Reply:
@jansel Just prefer to future proof it

Reply:
What is the case you are thinking about in the future?  This flag changes the behavior in the Triton repo and doesn't affect this repo.

Reply:
If the downstream callable is changed to take arguments etc. I just find people often forget to update typing signatures so the most specific, dynamic typing possible, the better (ie ParamSpec, TypeVar, etc.). If it's more specific, it will be more likely to flag the error in the future should some issue occur.

Reply:
There won't be any downstream callable since this function returns None.